mesonbuild/scripts/tags.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2019 The Meson development team
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import subprocess
+from pathlib import Path
+
+
+def list_files():
+    if os.path.exists('.git'):
+        return subprocess.run(['git', 'ls-tree', '-r', '--name-only', 'HEAD'],
+                              stdout=subprocess.PIPE).stdout
+
+    files = [str(p) for p in Path('.').glob('**/*')
+             if not p.is_dir() and
+             not next((x for x in p.parts if x.startswith('.')), None)]
+    return '\n'.join(files).encode()
+
+
+def cscope():
+    ls = '\n'.join(['"%s"' % f for f in list_files().decode().split()]).encode()
+    subprocess.run(['cscope', '-v', '-b', '-i-'], input=ls, check=True)
+
+
+def ctags():
+    ls = list_files()
+    subprocess.run(['ctags', '-L-'], input=ls, check=True)
+
+
+def etags():
+    ls = list_files()
+    subprocess.run(['etags', '-'], input=ls, check=True)
+
+
+def run(args):
+    tool_name = args[0]
+    srcdir_name = args[1]
+    os.chdir(srcdir_name)
+    assert tool_name in ['cscope', 'ctags', 'etags']
+    return globals()[tool_name]()
```

Comment:
This returns `None` as the exit value for the script. The tool functions should return the exit code of the call instead (and not use `check`). Something like:

```
   return subprocess.run(['etags', '-'], input=ls).returncode
```

Reply:
right, previous version was more correct. fixed.

Reply:
FWICT this has not been fixed. `cscope`, `ctags` and `etags` still don't return the returncode attribute. They return nothing, and thus the return value is `None`.

Reply:
I didn't push the new version